devDocs/startupShutdownInstall.md

```diff
@@ -0,0 +1,96 @@
+# NVDA Starting and Exiting outline
+
+## As a user, NVDA can be started by:
+
+1. For an installed copy:
+    1. Ctrl+Alt+N
+    1. Via secure desktop/ease of access
+    1. Via startup
+1. For an installed copy, portable copy, installer:
+    - By running the exe (see cli options too).
+    This can be triggered by a user or external process such as an existing NVDA instance
+    - test: `startupShutdownNVDA.Starts`
+1. For source: eg runnvda.bat
+
+## As a user, NVDA can be shutdown by:
+
+1. UI within NVDA, with and without an ExitDialog prompt (uses `triggerNVDAExit`):
+    1. NVDA+q
+        - test: `startupShutdownNVDA.Quits from keyboard, Restarts`
+    1. An unbound command to restart
+    1. A prompt to restart from changing certain settings
+    1. Via the NVDA menu -> Exit
+        - test: `startupShutdownNVDA.Quits from menu`
+1. A process sending `WM_QUIT`, eg a new NVDA process starting (uses `WM_QUIT`)
+1. A handled crash (directly causes a new process to start, terminates unsafely)
+    - test: `startupShutdownNVDA.Restarts on crash, Restarts on braille crash`
+1. An unhandled crash (terminates unsafely)
+1. An external command which kills the process (terminates unsafely) 
+1. Windows shutting down (terminates unsafely)
+    - On `wx.EVT_END_SESSION`, we save the config and play the exit sound. (Note: should this do more? NVDA may terminate unsafely directly after this)
+
+## Safe exit hooks/triggers
+
+### In process: `triggerNVDAExit`
+* A function in the core module
+* Only executes the code once, uses a lock and flag to ensure this
+* Uses a queue on the main thread to queue a safe shutdown
+* Once the queued shutdown starts:
+    1. the updateCheck is terminated
+    1. watchdog is terminated
+    1. addons and the brailleViewer are terminated, so we can close all windows safely
+    1. All NVDA windows are closed
+    1. Now that windows are closed, a new NVDA instance is started if requested
+
+
+### Out of process: `WM_QUIT`
+* a windows message which can be sent across processes which will force the main loop to exit and close most NVDA windows.
+* We subsequently run `triggerNVDAExit` to exit other required processes, and pump the queue to execute the shutdown.

```

Comment:
What does "other required processes" mean? Do you mean other copies of NVDA, or do you mean other subcomponants of NVDA?

Reply:
```suggestion
* We subsequently run `triggerNVDAExit` to ensure that clean up code isn't missed, and pump the queue to execute it.
```

Reply:
I just mean what `triggerNVDAExit` does, which I was trying to avoid from repeating.

Reply:
This reads better to me now, thanks.